src/core/src/GslMatrix.C

```diff
@@ -432,6 +441,50 @@ GslMatrix::chol()
   return iRC;
 }
 
+void
+GslMatrix::cholSolve(const GslVector & rhs, GslVector & sol) const
+{
+  queso_require_equal_to_msg(this->numCols(), rhs.sizeLocal(), "matrix and rhs have incompatible sizes");
+  queso_require_equal_to_msg(sol.sizeLocal(), rhs.sizeLocal(), "solution and rhs have incompatible sizes");
```

Comment:
Since the solution vector sol has to be pre-sized prior to call, consider documenting.